surface/frontend/Window1/main.js

```diff
@@ -29,4 +29,32 @@ module.exports = (where, socketHost) => {
     /*
         IPC Connection Section
     */
+
+    ipcRenderer.on('buddy-controls-from-win-1', (event, data) => {
+        where.setState({
+          directions: data
+        });
+    });
+
+
+    //updating the gamepad
+    setInterval(() => {
+      if(where.gp.ready === false) {
+//        console.log("not yet");
+        where.gp.selectController();
+      }
+      if((where.gp.ready === true) && (where.state.freeze === 0)) {
+        where.gp.update();
+//        console.log('success');
+} else if (where.state.freeze === 1){
+        this.gp.freeze();
```

Comment:
`Uncaught TypeError: Cannot read property 'freeze' of undefined`

Something's going wrong when I try to compile this